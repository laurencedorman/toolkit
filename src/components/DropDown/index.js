export { default } from './DropDown';
export { default as OptionContainer } from './OptionContainer';
