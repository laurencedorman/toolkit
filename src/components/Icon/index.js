export { default } from './Icon';
export { default as Svg } from './Svg';
export { default as Img } from './Img';
