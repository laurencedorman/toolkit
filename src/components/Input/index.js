/**
 * Copyright (c) Colibri SAS - ManoMano
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export { default } from './Input';
export { default as InputLight } from './InputLight';
