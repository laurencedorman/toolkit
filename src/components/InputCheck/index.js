export { default } from './InputCheck';
export { default as InputCheckGroup } from './InputCheckGroup';
