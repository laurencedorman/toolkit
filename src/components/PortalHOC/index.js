export { default as withPortal } from '../PortalHOC';
