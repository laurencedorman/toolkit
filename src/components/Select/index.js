export { default } from './Select';
export { default as WithReduxFormSelect } from './WithReduxFormSelect';
