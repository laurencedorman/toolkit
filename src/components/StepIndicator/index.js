export { default } from './StepIndicator';
export { default as Step } from './Step';
