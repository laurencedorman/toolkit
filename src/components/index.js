export { default as Alert } from './Alert';
export { default as Avatar } from './Avatar';
export { default as Button } from './Button';
export { default as Card } from './Card';
export { default as DropDown, OptionContainer } from './DropDown';
export { default as GetMeasure } from './GetMeasure';
export { default as HeightTransition } from './HeightTransition';
export { default as Hint } from './Hint';
export { default as Icon } from './Icon';
export { default as Iframe } from './Iframe';
export { default as Input, InputLight } from './Input';
export { default as InputCheck, InputCheckGroup } from './InputCheck';
export { default as Layout } from './Layout';
export { default as Loader } from './Loader';
export { default as Logo } from './Logo';
export { default as Modal } from './Modal';
export { default as Overlay } from './Overlay';
export { default as Portal } from './Portal';
export { default as Select, WithReduxFormSelect } from './Select';
export { default as Sidebar } from './Sidebar';
export { default as StepIndicator, Step } from './StepIndicator';
export { default as StepControler } from './StepControler';
export { default as Stepper } from './Stepper';
export { default as Switch } from './Switch';
export { default as Table } from './Table';
export {
  default as Tabs, Tab, TabBody, TabLabel,
} from './Tabs';
export { default as Tag } from './Tag';
export { default as Textarea } from './Textarea';
export { default as Toggle } from './Toggle';
export { default as ToggleDown } from './ToggleDown';
export { default as Tooltip } from './Tooltip';
export { default as Wrapper } from './Wrapper';
