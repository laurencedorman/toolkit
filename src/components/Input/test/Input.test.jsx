import React from 'react';
import { shallow } from 'enzyme';
import { Input } from 'components';

describe('Input', () => {
  const wrapper = shallow(
    <Input
      type="text"
      id="id"
      name="name"
      label="label"
      helper="helper"
    />,
  );

  it('should render without crash', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('should render a input', () => {
    expect(wrapper.find('input').length).toEqual(1);
  });

  it('should render a label', () => {
    expect(wrapper.find('label').text()).toEqual('label');
  });

  it('should render specific className', () => {
    wrapper.setProps({ value: 'value', valid: true });
  });

  it('should render specific className', () => {
    wrapper.setProps({ reverse: true, icon: true });
  });

  it('should render an icon', () => {
    wrapper.setProps({ valid: true, reverse: true });
  });

  it('should render an helper span', () => {
    expect(wrapper.find('span').text()).toEqual('helper');
  });

  it('should render an error span', () => {
    wrapper.setProps({ error: true });
  });
});
