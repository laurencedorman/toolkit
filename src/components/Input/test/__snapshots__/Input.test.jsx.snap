// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="content hasValue"
  >
    <input
      aria-hidden={false}
      aria-label="label"
      aria-required={false}
      className=""
      data-qa=""
      disabled={false}
      hidden={false}
      id="id"
      max={null}
      maxLength={null}
      min={null}
      minLength={null}
      name="name"
      onBlur={null}
      onChange={null}
      onFocus={null}
      onKeyDown={null}
      required={false}
      step={null}
      type="text"
      value="value"
    />
    <label
      className=""
      htmlFor="id"
    >
      label
    </label>
  </div>
  <span
    className="helper"
  >
    helper
  </span>
</div>
`;
